gemma-core/src/main/java/ubic/gemma/core/analysis/service/ExpressionDataFileServiceImpl.java

```diff
@@ -349,6 +352,21 @@ public void writeDiffExArchiveFile( BioAssaySet experimentAnalyzed, Differential
         }
     }
 
+    @Override
+    @Transactional(readOnly = true)
+    public void writeRawExpressionData( ExpressionExperiment ee, Writer writer ) throws IOException {
+        ee = expressionExperimentService.findWithRawExpressionDataVectors( ee );
+        if ( ee == null ) {
+            throw new IllegalArgumentException( "ExpressionExperiment has been removed." );
+        }
+        ExpressionDataDoubleMatrix matrix = expressionDataMatrixService.getRawExpressionDataMatrix( ee );
+        Set<ArrayDesign> ads = ee.getRawExpressionDataVectors().stream()
+                .map( RawExpressionDataVector::getDesignElement )
+                .map( CompositeSequence::getArrayDesign )
+                .collect( Collectors.toSet() );
+        new MatrixWriter().writeWithStringifiedGeneAnnotations( writer, matrix, getGeneAnnotationsAsStringsByProbe( ads ), true );
```

Comment:
Here I wanted to gather the gene metadata from the `GENE2CS` table, but that will wait due to the urgency of provding raw data access.